### DISCUSSION TITLE
Display attached enum documentation when hovering over an export field with a value

### DISCUSSION THREAD

**Discussion by SurWeaver** [godotengine/godot-proposals#9493] [Editor]
Below are examples based on the parent classes Node and Resource.

If I need to see documentation for a particular enumeration value, I won't see it in the inspector on nodes inherited from Node.
```gdscript
extends Node

##Enumeration documentation
enum Enumeration {
	## doc for A
	A,
	## doc for B
	B,
	## doc for C
	C,
}

## Enumeration value documentation
@export var enum_value: Enumeration
```
![image](https://github.com/godotengine/godot-proposals/assets/47503650/fc4d6998-0e9a-4ae3-b980-bb123443c4b6)

Hovering the cursor over the value itself will not display anything in the inspector. This field does not have a tooltip.

Another example is the Resource class, whose export fields play an important role.

The code for a resource example is almost identical:
```gdscript
extends Resource
class_name TestResource

##Enumeration documentation
enum Enumeration {
	## doc for A
	A,
	## doc for B
	B,
	## doc for C
	C,
}

## Enum value documentation
@export var enum_value: Enumeration

## Int value documentation
@export var int_value: int
```

![image](https://github.com/godotengine/godot-proposals/assets/47503650/076c7994-624d-46a0-81e6-5d61107289d4)

As you can see in the image above, here the documentation is displayed for the field itself, as well as for all the values of the enumeration.

While the value field itself does not display either the value or the documentation for it.
This can make the screen very busy if there are a lot of values and all of them are accompanied by documentation.
People working on filling the game with resources could conveniently view the documentation for each value if a doc tooltip was displayed by hovering over the value field itself or over an item in the dropdown menu.

![image](https://github.com/godotengine/godot-proposals/assets/47503650/d5140742-9c27-4ac4-9b92-61b407d48f04)

At the same time there is a tooltip for numbers.
![image](https://github.com/godotengine/godot-proposals/assets/47503650/7d1be0d7-6f5c-4131-bb68-aa270bbd68a1)

It would be very convenient if the tooltip for an enumeration contained the name of the type, the text value, its numeric representation, and documentation.
For example:
```
type: Enumeration
doc for A
0: A
```

**Comment by dalexeev** [comment]
* See godotengine/godot#90488.

**Reply by SurWeaver** [reply]
I didn't mention it, but yeah, I'm currently encountering a bug that doesn't even display that description. As a temporary solution, adding `@tool` to the resource script and the script containing the enumeration helped me.
Thanks for the quick fix!

And what do you think about adding an enum tooltip specifically to the right field and dropdown items?

**Reply by SurWeaver** [reply]
I have another additional question if you please.

I have a Resource script (with class_name CardPersonalInfo) with two fields on two different enumerations. These enums lie in two different scripts that have class_name (Action, Effect).
When I load the editor, I open the created resource with CardPersonalInfo. It loads all the values correctly in the dropdown menu, but the documentation is not found. Even taking into account that all scripts have class_name.
When I change something in the enum scripts so it gives an error and change it back, all the documentation for both fields loads.

This situation is floating around for me, as sometimes it works from the start of the editor, sometimes it doesn't.
If it matters, the resource script is `@tool`.
